admin/code/CMSBatchActionHandler.php

```diff
@@ -179,7 +179,11 @@ protected function actionByName($name) {
 		// Find the action handler
 		$actions = $this->batchActions();
 		if(!isset($actions[$name]['class'])) {
-			throw new InvalidArgumentException("Invalid batch action: {$name}");
+			$action = $actions[$name];
+
+			throw new InvalidArgumentException(
+				require __DIR__ . '/../templates/Exceptions/InvalidBatchActionException.php'
```

Comment:
I'd be more comfortable with an absolute path `require FRAMEWORK_DIR . '/templates/Exceptions/InvalidBatchActionException.php'`

FRAMEWORK_DIR is defined somewhere very close to the start of the boot sequence, so should be safe to use here.